Neos.ContentRepository/Classes/Domain/Repository/NodeDataRepository.php

```diff
@@ -367,6 +368,44 @@ public function findOneByIdentifier($identifier, Workspace $workspace, array $di
         return null;
     }
 
+    /**
+     * Find all objects and return an IterableResult
+     *
+     * @return IterableResult
+     */
+    public function findAllIterator()
+    {
+        /** @var \Doctrine\ORM\QueryBuilder $queryBuilder */
+        $queryBuilder = $this->entityManager->createQueryBuilder();
+        return $queryBuilder
+            ->select('node')
+            ->from($this->getEntityClassName(), 'node')
+            ->getQuery()->iterate();
+    }
+
+    /**
+     * Iterate over an IterableResult and return a Generator
+     *
+     * This method is useful for batch processing a huge result set.
+     *
+     * @param IterableResult $iterator
+     * @param callable $callback
+     * @return \Generator
```

Comment:
That is not really true, is it? The method returns `NodeData`.

Reply:
I thought so too at first, but technically a Generator is returned. PhpStorm complains about NodeData as return type, If I would ignore that and define the return type:

`public function iterate(IterableResult $iterator, callable $callback = null): NodeData`

I get

```
PHP Fatal error:  Generators may only declare a return type of Generator, Iterator, Traversable, or iterable, Neos\ContentRepository\Domain\Model\NodeData is not permitted in /var/www/Packages/Application/Neos.ContentRepository/Classes/Domain/Repository/NodeDataRepository.php on line 395
```

Besides that. We have iterators in quite some Repositories (AssetRepository, ResourceRepository ... ) - what do you think about adding a findAllIterator and magic findByOBJECTIterator directly to the `Neos\Flow\Persistence\Repository` ?

Reply:
TIL:

> technically a Generator is returned

Ugh, that's ugly IMHO. But anyway, that's life.

The `findAllIterator` idea sounds good… but I don't understand what you mean with "magic `findByOBJECTIterator`"…